include/aws/common/math.h

```diff
@@ -25,179 +16,61 @@
  * permissions and limitations under the License.
  */
 
-/**
- * Multiplies a * b. If the result overflows, returns 2^64 - 1.
- */
-#if (AWS_ENABLE_HW_OPTIMIZATION && (defined(_X86_64) || defined(_M_X64) || defined(_M_ARM)))
-AWS_STATIC_IMPL uint64_t aws_mul_u64_saturating(uint64_t a, uint64_t b) {
-#    if defined(__x86_64__) && (defined(__GNUC__) || defined(__clang__))
-    /* We can use inline assembly to do this efficiently on x86-64 and x86.
-
-    we specify rdx as an output, rather than a clobber, because we want to
-    allow it to be allocated as an input register */
-
-    uint64_t rdx;
-    __asm__("mulq %q[arg2]\n" /* rax * b, result is in RDX:RAX, OF=CF=(RDX != 0) */
-            "cmovc %q[saturate], %%rax\n"
-            : /* in/out: %rax = a, out: rdx (ignored) */ "+&a"(a), "=&d"(rdx)
-            : /* in: register only */ [arg2] "r"(b),
-              /* in: saturation value (reg/memory) */ [saturate] "rm"(~0LL)
-            : /* clobbers: cc */ "cc");
-    (void)rdx; /* suppress unused warnings */
-    return a;
-#    elif defined(_M_X64) || defined(_M_ARM)
-    uint64_t out;
-    uint64_t ret_val = _umul128(a, b, &out);
-    return out == 0 ? ret_val : ~(uint64_t)0;
-#    else
-#        error "not supported."
-#    endif
-}
+#include <aws/common/common.h>
+#include <aws/common/config.h>
 
-/**
- * Multiplies a * b and returns the truncated result in *r. If the result
- * overflows, returns 0, else returns 1.
- */
-AWS_STATIC_IMPL int aws_mul_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
-#    if defined(__x86_64__) && (defined(__GNUC__) || defined(__clang__))
-    /* We can use inline assembly to do this efficiently on x86-64 and x86. */
-
-    int flag;
-    uint64_t result = a;
-    __asm__("mulq %q[arg2]\n"    /* rax * b, result is in RDX:RAX, OF=CF=(RDX != 0) */
-            "setno %%dl\n"       /* rdx = (OF = 0) */
-            "and $0xFF, %%edx\n" /* mask out flag */
-            : /* in/out: %rax (with first operand) */ "+&a"(result), "=&d"(flag)
-            : /* in: reg for 2nd operand */
-            [arg2] "r"(b)
-            : /* clobbers: cc */ "cc");
-    *r = result;
-    return flag;
-#    elif defined(_M_X64) || defined(_M_ARM)
-    uint64_t out;
-    *r = _umul128(a, b, &out);
-
-    return out == 0;
-#    else
-#        error "Not supported."
-#    endif
-}
+#include <stdlib.h>
+
+#ifdef __cplusplus
+extern "C" {
```

Comment:
[optional] you can replace this whole block with `AWS_EXTERN_C_BEGIN` and the close with `AWS_EXTERN_C_END`